bodhi/server/validators.py

```diff
@@ -427,6 +428,43 @@ def validate_acls(request, **kwargs):
                            'unable to determine ACLs.')
         return
 
+    # Allow certain groups to push updates for any package
+    admin_groups = config['admin_packager_groups']
+    for group in admin_groups:
+        if group in user_groups:
+            log.debug(f'{user.name} is in {group} admin group')
+            return
+
+    # Make sure the user is in the mandatory packager groups. This is a
+    # safeguard in the event a user has commit access on the ACL system
+    # but isn't part of the mandatory groups.
+    mandatory_groups = config['mandatory_packager_groups']
+    for mandatory_group in mandatory_groups:
+        if mandatory_group not in user_groups:
+            error = (f'{user.name} is not a member of "{mandatory_group}", which is a '
+                     f'mandatory packager group')
+            request.errors.add('body', 'builds', error)
+            return
+
+    # If we try to create or edit a side-tag update, check if user owns the side-tag
+    if 'from_tag' in request.validated and request.validated.get('from_tag') is not None:
```

Comment:
would `if request.validated.get('from_tag') is not None` be enough here ? Why do we need to check that "from_tag" is in request.validated ?

Reply:
Yeah, you're right. The Schema provides a `missing=None` directive, so I think it's safe to suppose that parameter is always present in the request.